### ISSUE TITLE
Feature Request: DevTestLabs

### ISSUE THREAD

**Issue by tombuildsstuff** [terraform-providers/terraform-provider-azurerm#132]
Requested [on the Terraform Mailing List](https://groups.google.com/forum/#!topic/terraform-tool/79KsLdQtZ2Q)

Taking a quick look into this - [it's supported in the SDK](https://github.com/Azure/azure-sdk-for-go/tree/master/arm/devtestlabs) and would be multiple resources, which a subset of are listed below:

- `azurerm_devtestlabs_lab`
- `azurerm_devtestlabs_schedule`
- `azurerm_devtestlabs_secret`
- `azurerm_devtestlabs_formula`
- `azurerm_devtestlavs_claimable_virtual_machine`
- `azurerm_devtestlabs_virtual_machine`

**Comment by ivadim** [comment]
Hi @tombuildsstuff, any updates on this?

**Comment by neidiom** [comment]
@tombuildsstuff if I understand this feature would add support for VM auto shutdown? What is the progress on this?

**Comment by HighwayofLife** [comment]
This is also useful for Jumpboxes in Prod, which don't need to be on all the time, and could increase security if we had the ability to shut them down. This can be done through the Azure CLI, but would be better if we had schedules available in Terraform.

**Comment by A140475** [comment]
Waiting on this feature also. I have my whole VM Terraformed and then have to go into the portal to manually schedule a shutdown, :)

**Comment by brs-rom** [comment]
Guys, do you have roadmap no this future? Any estimation?

**Comment by tombuildsstuff** [comment]
👋

Initial support for DevTest Lab's (and Virtual Networks) has been merged in #1944 which will be available in v1.16 of the AzureRM Provider in the near future. Since this is a combined issue - please open a separate issue if there's a particular component of DevTest Labs which you're interested in Terraform supporting.

Thanks!

**Comment by tombuildsstuff** [comment]
👋

Just to let you know that support for this has now landed in v1.16 of the AzureRM Provider (here's the full changelog](https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.16.0/CHANGELOG.md)). You can upgrade to to this version by specifying it in your provider block:

```
provider "azurerm" {
  version = "=1.16.0"
}
```

... and then running `terraform init -upgrade` :)

Thanks!
